component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPKeyGenerator.java

```diff
@@ -146,9 +151,13 @@ public static Map<String, String> generateClaimsForFedUser(String username, Stri
         try {
             TOTPAuthenticatorKey key = generateKey(tenantDomain, context);
             secretKey = key.getKey();
-            claims.put(TOTPAuthenticatorConstants.SECRET_KEY_CLAIM_URL,
-                    TOTPUtil.encrypt(secretKey));
-
+            Map<String, String> claimProperties = TOTPUtil.getClaimProperties(tenantDomain,
```

Comment:
"Map<String, String> claimProperties = TOTPUtil.getClaimProperties(tenantDomain,
                    TOTPAuthenticatorConstants.SECRET_KEY_CLAIM_URL);
            if (claimProperties.get("EnableEncryption") != null) {
                claims.put(TOTPAuthenticatorConstants.SECRET_KEY_CLAIM_URL, secretKey);
            } else {
                claims.put(TOTPAuthenticatorConstants.SECRET_KEY_CLAIM_URL, TOTPUtil.encrypt(secretKey));
            }"

This complete clock is a common logic which can be included into a private method. Isn't it ?

Reply:
This change will be reverted after the migration is completed